.circleci/config.yml

```diff
@@ -0,0 +1,58 @@
+version: 2
+
+docker_sbt: &docker_sbt
+  docker:
+    - image: circleci/openjdk:8-jdk
+      environment:
+        DEBIAN_FRONTEND: noninteractive
+        SBT_OPTS: -Xms256m -Xmx756G -XX:+CMSClassUnloadingEnabled -XX:+UseSerialGC -XX:MaxMetaspaceSize=2G
+        DISABLE_SCALAFMT: true
+  working_directory: ~/repo
+  environment:
+    JVM_OPTS: -Xmx3200m
+
+cache_save: &cache_save
+  save_cache:
+    paths:
+      - ~/.m2
+      - ~/.sbt
+      - ~/.ivy2/cache
+    key: v1-{{ checksum "build.sbt" }}
+
+cache_restore: &cache_restore
+  restore_cache:
+    keys:
+      - v1-{{ checksum "build.sbt" }}
+      - v1-
+jobs:
+  release:
+    <<: *docker_sbt
+    steps:
+      - checkout
+      - *cache_restore
+      - run:
+          name: Release
+          command: sbt ';set version in ThisBuild := "'`git describe`'"; test; publish'
```

Comment:
I'd recommend to publish a snapshot version for each merged PR and have an additional (manually triggered) stage in the workflow, which does a full version release. Using `sbt-release` would be also preferable in my opinion, as you could automate the whole release without having to create the tags yourself. I'd also recommend to auto increment the minor version on release.

Reply:
The problem with sbt release is we'll want to increase the version differently depending on what we've done.

We _could_ publish a snapshot version but I don't really see the value of it given that we won't want to ever depend on them

Reply:
> The problem with sbt release is we'll want to increase the version differently depending on what we've done.

Sure, so a sensible strategy is to bump up the minor version and make it a snapshot after the release, but if you want to release a patch or a major version all you have to do is commit a version change before making the release.

> We could publish a snapshot version but I don't really see the value of it given that we won't want to ever depend on them

Snapshots might be useful for testing the changes in your projects depending on this library before making a release, but you can achieve the same thing by publishing the snapshot locally, so yeah if you don't see this being usefull then that's fair enough.